src/uct/ugni/rdma/ugni_rdma_iface.c

```diff
@@ -0,0 +1,272 @@
+/**
+ * Copyright (c) UT-Battelle, LLC. 2014-2015. ALL RIGHTS RESERVED.
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2014.  ALL RIGHTS RESERVED.
+ * $COPYRIGHT$
+ * $HEADER$
+ */
+
+#include <pmi.h>
+#include "ucs/type/class.h"
+#include "uct/tl/context.h"
+
+#include <uct/ugni/base/ugni_iface.h>
+#include "ugni_rdma_iface.h"
+#include "ugni_rdma_ep.h"
+
+static ucs_config_field_t uct_ugni_rdma_iface_config_table[] = {
+    /* This tuning controls the allocation priorities for bouncing buffers */
+    { "", "MAX_SHORT=2048;MAX_BCOPY=2048;ALLOC=huge,mmap,heap", NULL,
+    ucs_offsetof(uct_ugni_iface_config_t, super), UCS_CONFIG_TYPE_TABLE(uct_iface_config_table)},
+
+    UCT_IFACE_MPOOL_CONFIG_FIELDS("FMA", -1, 0, "fma",
+                                  ucs_offsetof(uct_ugni_iface_config_t, mpool),
+                                  "\nAttention: Setting this param with value != -1 is a dangerous thing\n"
+                                  "and could cause deadlock or performance degradation."),
+
+    {NULL}
+};
+
+static ucs_status_t uct_ugni_query_tl_resources(uct_pd_h pd,
```

Comment:
The function is expected to be the same for rdma and all the rest. Shell we move it to base ?